### ISSUE TITLE
Instantiated initial value if null

### ISSUE THREAD

**Issue by henk-lim** [rrousselGit/freezed#601]
**Is your feature request related to a problem? Please describe.**
I'm trying to implement [IsarLinks](https://isar.dev/links.html#isarlinks) with Freezed. This requires a field to be instantiated and initialised during the construction, which is not allowed by Freezed.

Currently the only way to define a default value is by using the @Default(literal) annotation. I've searched through issues around this which dated back to 2 years - #64, #82, #149.  These issues imply that the problem is gated by JsonSerializable. Which it is, if we are looking at it from the perspective of "default" values.

**Describe the solution you'd like**

What about an initial value? Why not add an @Initial(value) annotation?

So the generated code would then become something like this:

`Constructor({Class? field}) : field = field ?? intialValue, super._();`

instead of this:

`Constructor({this.field = defaultValue}) : super._();`

**Describe alternatives you've considered**

I tried using this [hack](https://github.com/rrousselGit/freezed/issues/82#issuecomment-593784716) by by adding a JsonKey() to the field. But this doesn't work because we are still working in the realm of Default value which only accepts const literal.

**Additional context**

This problem is not related to JsonSerializable as far as I'm concerned because I've been able to address this issue by using a combination of packages to replace Freezed - Equatable, CopyWithExtension, and JsonSerialiazable. Not ideal because of the convenience of Freezed having all the features in one package. However, I've come up with this problem too many times needing an initial value which Freezed does not support.

**Comment by rrousselGit** [comment]
Closing in favor of https://github.com/rrousselGit/freezed/issues/64
